### PR TITLE
Jdp230102 20 - Entity User Tests

### PR DESCRIPTION
**Pull request by viepovsky**

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 1]
```diff
@@ -8,16 +8,17 @@
...
 
     @Id
     @NonNull
+    @GeneratedValue
     @Column(name = "ID", unique = true)
-    @GeneratedValue(strategy = GenerationType.AUTO)
     private Long id;
 
-    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @OneToOne(fetch = FetchType.LAZY)
```
Comment (andwis93) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java]: It looks like relations @OneToOne should be EAGER according to this article: 

https://nullpointerexception.pl/trzy-rzeczy-ktore-powinienes-wiedziec-o-hibernate/

so I would leave it EAGER

File: src/main/java/com/kodilla/ecommercee/domain/Session.java [hunk 2]
```diff
@@ -8,25 +8,27 @@
...
     @NotNull
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    @Column(name ="SESSION_ID", unique = true)
-    private long sessionId;
+    @GeneratedValue
+    @Column(name ="ID", unique = true)
+    private long id;
 
-    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    @ManyToOne(fetch = FetchType.LAZY)
```
Comment (andwis93) [inline src/main/java/com/kodilla/ecommercee/domain/Session.java]: According to this article @ManyToOne is EAGER as well and @OneToMany should be LAZY:

https://nullpointerexception.pl/trzy-rzeczy-ktore-powinienes-wiedziec-o-hibernate/

That was actually my mistake. I should put EAGER at the first place

File: src/main/java/com/kodilla/ecommercee/repository/UserRepository.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Transactional
+@Repository
+public interface UserRepository extends CrudRepository<User, Long> {
+    List<User> findAll();
```
Comment (andwis93) [inline src/main/java/com/kodilla/ecommercee/repository/UserRepository.java:13]: Right now I don't remember why (I will have to check it) but we always used annotation @Override if we implement methods that already are implemented in CrudRepository.

**Review by andwis93 — APPROVED** [review]

File: src/main/java/com/kodilla/ecommercee/repository/UserRepository.java (hunk 3 above)
Comment (viepovsky) [reply src/main/java/com/kodilla/ecommercee/repository/UserRepository.java:13]: Yes, that was written in kodilla course but I don't use it, it works well without it. If you want I can add this.

File: src/main/java/com/kodilla/ecommercee/domain/Session.java (hunk 2 above)
Comment (viepovsky) [reply src/main/java/com/kodilla/ecommercee/domain/Session.java]: Ok, I've changed files as you written.

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 1 above)
Comment (viepovsky) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java]: Ok, I've changed files as you written.

**Review by MMacina — APPROVED** [review]

**Pull request merged**